dms/models/dms_access_groups.py

```diff
@@ -39,74 +49,100 @@ class DmsAccessGroups(models.Model):
-        relation="dms_directory_groups_rel",
         string="Directories",
-        column1="gid",
-        column2="aid",
-        auto_join=True,
-        readonly=True,
+        comodel_name="dms.directory",
+        readonly=False,
     )
     complete_directory_ids = fields.Many2many(
+        string="Complete directories",
         comodel_name="dms.directory",
         relation="dms_directory_complete_groups_rel",
         column1="gid",
         column2="aid",
-        string="Complete directories",
         auto_join=True,
         readonly=True,
     )
     count_users = fields.Integer(compute="_compute_users", store=True)
     count_directories = fields.Integer(compute="_compute_count_directories")
     parent_group_id = fields.Many2one(
-        comodel_name="dms.access.group",
         string="Parent Group",
+        comodel_name="dms.access.group",
         ondelete="cascade",
         index="btree",
     )
-
     child_group_ids = fields.One2many(
+        string="Child Groups",
         comodel_name="dms.access.group",
         inverse_name="parent_group_id",
-        string="Child Groups",
     )
     group_ids = fields.Many2many(
+        string="Groups",
         comodel_name="res.groups",
         relation="dms_access_group_groups_rel",
         column1="gid",
         column2="rid",
-        string="Groups",
     )
     explicit_user_ids = fields.Many2many(
+        string="Explicit Users",
         comodel_name="res.users",
         relation="dms_access_group_explicit_users_rel",
         column1="gid",
         column2="uid",
-        string="Explicit Users",
     )
     users = fields.Many2many(
+        string="Group Users",
         comodel_name="res.users",
         relation="dms_access_group_users_rel",
         column1="gid",
         column2="uid",
-        string="Group Users",
         compute="_compute_users",
         auto_join=True,
         store=True,
         recursive=True,
     )
 
+    # ----------------------------------------------------------
+    # SQL Constraints
+    # ----------------------------------------------------------
+    @api.constrains("parent_path")
+    def _check_parent_recursiveness(self):
+        """
+        Forbid recursive relationships.
+        """
+        for one in self.filtered("parent_group_id"):
+            if str(one.id) in one.parent_path.split("/"):
+                raise ValidationError(
+                    _("Parent group '%(parent)s' is child of '%(current)s'.")
+                    % {
+                        "parent": one.parent_group_id.display_name,
+                        "current": one.display_name,
+                    }
+                )
+
+    # ----------------------------------------------------------
+    # Default methods
+    # ----------------------------------------------------------
+    @api.model
+    def default_get(self, fields_list):
+        """
+        When creating a new group, add the current user to the explicit users.
+        """
```

Comment:
Incorrect PEP257 docstring format. It can be one line, but if it doesn't fit, it should be this way:
```suggestion
        """When creating a new group, add the current user to the explicit users.
        """
```

Reply:
Does that mean that you want me to re-write every docstring of every method?